modules/hal_nordic/nrf_802154/radio/platform/nrf_802154_random_zephyr.c

```diff
@@ -8,6 +8,10 @@
 #include <platform/nrf_802154_temperature.h>
 #include <zephyr/drivers/entropy.h>
 
+/* Ensure that counter driver for TIMER1 is not enabled. */
+BUILD_ASSERT(DT_NODE_HAS_STATUS(DT_NODELABEL(timer1), disabled),
```

Comment:
I don't understand why this change is needed here. Could you please explain?

Reply:
So there is a counter API which is generic API to be used for TIMER and RTC. The configuration for counter instances used to be split between Kconfig and DT so to use one instance you had to enable it in Kconfig.
There are cases when TIMER or RTC is used outside of counter API (system timer, radio driver or bluetooth) in that case we need to protect against using same instance twice, e.g. if radio driver is used TIMER1 cannot be used for counter.

We used to have Kconfig flags for that but after moving counter configuration to DT they are useless.

So assert checks if timer1 node is disabled when radio driver is used.

Reply:
> if radio driver is used TIMER1 cannot be used for counter

That's reasonable, but in my opinion it's messy to put this check in a completely unrelated platform file that implements random API for the 802.15.4 driver. If anyone creates their own implementation of this API, for example using a different PRNG, they will lose this check. Wouldn't it be better to introduce this dependency in 802.15.4 driver's Kconfig by adding something like:
```
depends on $(dt_nodelabel_has_status,timer1,disabled)
```
or similar (not sure if the syntax I used is correct)?

Reply:
LGTM, @e-rk what do you think?

Reply:
My opinion is that having the check as @jciupis suggested would be preferable. That way we stay consistent with the rest of the system. The lack of information is the problem of Kconfig itself and I believe that it should be addressed there. If every module ended up with it's own dependency checking that is parallel to Kconfig we would end up in a mess.

Reply:
Ok, i pushed take 2 where i used Kconfig approach. If there is a resource conflict then compilation fails without any verbal explanation.